cli_tools/gce_windows_upgrade/main.go

```diff
@@ -0,0 +1,99 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed targetOS in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// gce_windows_upgrade is a tool for upgrading GCE Windows instances.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/logging/service"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_windows_upgrade/upgrader"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+)
+
+var (
+	// project/zone/network/subnet/labels are not required when the upgrade has targetOS happen in place.
+	clientID               = flag.String(upgrader.ClientIDFlagKey, "", "Identifies the client of the importer, e.g. `gcloud` or `pantheon`.")
+	instance               = flag.String("instance", "", "Instance targetOS upgrade, in the form of 'projects/<project>/zones/<zone>/instances/<instance>'.")
```

Comment:
Only gcloud. We don't expect the wrapper to be used directly by customer. Even that happens, the rich experience of supporting multiple formats is not important at this layer. Will convert whatever input to URI at gcloud layer.

Reply:
With other wrappers we document API via Cloud Build. It's possible we'll do that for windows upgrade as well for customers who want to start upgrades programmatically. Windows upgrader should not assumed it will be called only from gcloud. Hence, supporting short names on upgrader level is a good idea to future proof it for API use.

Reply:
I checked other APIs' similar field. As you said, many of them support short name. I just added the support to both URI and simple name.